src/rust/engine/src/nodes.rs

```diff
@@ -368,10 +371,13 @@ impl ExecuteProcess {
         .map(|(name, dest)| Ok((CacheName::new(name)?, RelativePath::new(dest)?)))
         .collect::<Result<_, String>>()?;
 
-    let jdk_home = externs::getattr_as_optional_string(value, "jdk_home").map(PathBuf::from);
+    // Returns PyResult<Option<String>
+    let jdk_home = externs::getattr_as_optional_string(value, "jdk_home")
+      .unwrap()
+      .map(PathBuf::from);
 
     let execution_slot_variable =
-      externs::getattr_as_optional_string(value, "execution_slot_variable");
+      externs::getattr_as_optional_string(value, "execution_slot_variable").unwrap_or(None);
```

Comment:
Rather than `unwrap_or(None)` (which would have the effect of hiding the attribute error), this should be something roughly like:
```suggestion
      externs::getattr_as_optional_string(value, "execution_slot_variable")
        .map_err(|e| format!("Failed to get the `execution_slot_variable` field: {e}"))?;
```